catalog/spatial/csw/spatial-csw-source-common/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/common/source/AbstractCswSource.java

```diff
@@ -947,6 +959,23 @@ public ResourceResponse retrieveResource(URI resourceUri,
         return resourceReader.retrieveResource(resourceUri, requestProperties);
     }
 
```

Comment:
For the most part, yes. The ReliableResourceDownloader was modified to expect `retrieveResource(bytesSkipped)` to return a resource that has an input stream which begins with the offset of `bytesSkipped`, and does not care whether or not the server it's requesting the product from supports range-headers. The URLResourceReader (and `GetRecordsMessageBodyReader`) now handles all the logic for sending range requests and modifying the input stream based on whether Partial Content responses are supported or not.

Aside from `GetRecordsMessageBodyReader` though, I believe that everything else delegates to the `URLResourceReader` to actually make the get request and open the input stream for the resource.